agent/api/task/task_linux.go

```diff
@@ -148,6 +148,14 @@ func (task *Task) BuildLinuxResourceSpec(cGroupCPUPeriod time.Duration) (specs.L
 		linuxResourceSpec.Memory = &linuxMemorySpec
 	}
 
+	// Set task pids limit if set via ECS_TASK_PIDS_LIMIT env var
+	if taskPidsLimit > 0 {
+		pidsLimit := &specs.LinuxPids{
+			Limit: int64(taskPidsLimit),
+		}
+		linuxResourceSpec.Pids = pidsLimit
```

Comment:
`linuxResourceSpec.Pids = pidsLimit`
As a preview, will this use the `--pids-limit` via docker api?  Is this also something we can do via contianerd directly?

Reply:
also what limits should we set on this? is there an upper bound enforced anywhere?

Reply:
It seems we create cgroup resources by ourselves - https://github.com/aws/amazon-ecs-agent/blob/master/agent/taskresource/cgroup/cgroup.go#L269-L283

Reply:
correct, task-level cgroup is created ourselves. We use the containerd/cgroups for this already so the implementation is easy to port (https://github.com/containerd/cgroups)